parcels/fieldset.py

```diff
@@ -198,7 +172,7 @@ def from_netcdf(cls, filenames, variables, dimensions, indices={},
             dims['data'] = name
             inds = indices[var] if var in indices else indices
 
-            fields[var] = Field.from_netcdf(var, dims, paths, inds, units=units[var],
```

Comment:
@erikvansebille 
If I am not wrong, this units argument was previously ignored. This now fixed in this version that reads the grid which has a mesh argument